host.go

```diff
@@ -333,6 +373,17 @@ func (h *gpbftRunner) BroadcastMessage(msg *gpbft.GMessage) error {
 		log.Errorw("appending to WAL", "error", err)
 	}
 
+	h.msgsMutex.Lock()
+	if h.selfMessages[msg.Vote.Instance] == nil {
+		h.selfMessages[msg.Vote.Instance] = make(map[roundPhase][]*gpbft.GMessage)
+	}
+	key := roundPhase{
+		round: msg.Vote.Round,
+		phase: msg.Vote.Phase,
+	}
+	h.selfMessages[msg.Vote.Instance][key] = append(h.selfMessages[msg.Vote.Instance][key], msg)
```

Comment:
This is called by re-broadcast, which will exponentially increase the number of messages.

Reply:
We are also WALing the re-boradcasted messages, which is probably not what we want to do. I suggest to have a different path for sending out re-broadcast messages that only checks the equivocation filter.